docs/source/examples/Widget Custom.ipynb

```diff
@@ -0,0 +1,785 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "nbsphinx": "hidden"
+   },
+   "source": [
+    "[Index](Index.ipynb) - [Back](Widget Styling.ipynb) - [Next](Widget Asynchronous.ipynb)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "# Building a Custom Widget -  Email widget"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The widget framework is built on top of the Comm framework (short for communication).  The Comm framework is a framework that allows the kernel to send/receive JSON messages to/from the front end (as seen below).\n",
+    "\n",
+    "![Widget layer](images/WidgetArch.png)\n",
+    "\n",
+    "To create a custom widget, you need to define the widget both in the browser and in the Python kernel.\n",
```

Comment:
This section is nice, but feels a bit out of place at the top of the page. What do you think about moving it to the bottom? Then it could be a "How did this all work" section.

Reply:
Perhaps also link to the widget low level notebook that goes into far more detail.

Reply:
Moving this block a bit below.